src/main/java/bio/terra/cli/command/resource/create/AwsStorageFolder.java

```diff
@@ -0,0 +1,60 @@
+package bio.terra.cli.command.resource.create;
+
+import bio.terra.cli.command.shared.WsmBaseCommand;
+import bio.terra.cli.command.shared.options.ControlledResourceCreation;
+import bio.terra.cli.command.shared.options.Format;
+import bio.terra.cli.command.shared.options.WorkspaceOverride;
+import bio.terra.cli.serialization.userfacing.input.CreateAwsStorageFolderParams;
+import bio.terra.cli.serialization.userfacing.input.CreateResourceParams;
+import bio.terra.cli.serialization.userfacing.resource.UFAwsStorageFolder;
+import bio.terra.workspace.model.StewardshipType;
+import picocli.CommandLine;
+
+/**
+ * This class corresponds to the fourth-level "terra resource create aws-storage-folder" command.
+ */
+@CommandLine.Command(
+    name = "aws-storage-folder",
+    description = "Add a controlled AWS storage folder.",
+    showDefaultValues = true)
+public class AwsStorageFolder extends WsmBaseCommand {
+  @CommandLine.Mixin ControlledResourceCreation controlledResourceCreationOptions;
+  @CommandLine.Mixin WorkspaceOverride workspaceOption;
+  @CommandLine.Mixin Format formatOption;
+
+  @CommandLine.Option(
+      names = "--location",
+      defaultValue = "us-east-1",
+      description =
+          "The AWS location of the storage folder (https://docs.aws.amazon.com/general/latest/gr/s3.html).")
+  private String location;
+
```

Comment:
For other resources, we let users specify a Terra resource name (e.g. to display in the UI) that's different from the underlying cloud resource's name, like here https://github.com/DataBiosphere/terra-cli/blob/main/src/main/java/bio/terra/cli/command/resource/create/GcsBucket.java#L38.

I missed that S3 folders don't take that as a separate input parameter in the WSM API so there's not much you can do here until we change that, though

Reply:
This totally escaped me during review as well.  Let's stick with it for this PR, but I'll create follow on Jira tasks to double back and separate these in the API so that prefix need not match resource name.

Reply:
I had come to assume that this was by design, noted that resource name may be different from prefix name. 
created https://verily.atlassian.net/browse/TERRA-523

Reply:
Resolving this comment, since it it tracked elsewhere and this is not going to be fixed in this PR